src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -64,7 +69,26 @@ def resolve(self, root_reqs, check_supported_wheels):
             self.factory.make_requirement_from_install_req(r)
             for r in root_reqs
         ]
-        self._result = resolver.resolve(requirements)
+
+        try:
+            self._result = resolver.resolve(requirements)
+        except ResolutionImpossible as exc:
+            # TODO: This is just an initial version. May need more work.
+            # Also could do with rewriting to fit better into 80-char
+            # lines :-(
+            for req, parent in exc.causes:
+                logger.critical(
+                    "Could not find a version that satisfies " +
+                    "the requirement " +
+                    str(req) +
+                    ("" if parent is None else " (from {})".format(
+                        parent.name
+                    ))
+                )
```

Comment:
Does this output something like:

```
Could not find a version that satisfies the requirement Django>2.0 (...)
Could not find a version that satisfies the requirement Django<=1.11 (...)
```

This could feel confusing to users :| Maybe we can do:

```
Could not find a version that satisfies the requirement:
    Django>2.0 (...)
    Django<=1.11 (...)
```

Reply:
It might, and yes that would probably be better. My initial concern was mainly to not get a traceback for `pip install misspelled_thing`, preferably keeping the output as near to the current output as we can (we don't have the information in the exception to include "(from versions: xxx)" at the moment).

I'm happy to iterate on this - I guess my main question right now is whether we want to get *something* into the beta, or if we can live with "proper messages" being something for the post-20.1 improvements?

Also, the question of how closely we want to match the current output ties into this. @ei8fdb was pretty strongly of the view that we should try to get as close as possible, so I was working on trying to replicate current output - and that's a case I haven't been able to replicate (probably because the current resolver's "first come, first served" basis means that it never hits that situation.

Reply:
I am definitely in favour of having something for the release. This PR as-is would br much better than a cruel `ResolutionImpossible`.